editor/src/components/editor/editor-component.tsx

```diff
@@ -418,11 +408,46 @@ export const EditorComponentInner = React.memo((props: EditorProps) => {
         <ModalComponent />
         <ToastRenderer />
         <LockedOverlay />
+        <GithubDataManager />
       </SimpleFlexRow>
     </>
   )
 })
 
+const GithubDataManager = () => {
```

Comment:
I think it would be worth it to call this `useGithubData` or similar, because it uses hooks inside, and if this is used in an `if` or after an early return the app will break with React errors (as outlined in https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) (and the linter can help enforce these rules if the function's name starts with `use...`

Reply:
yeah, but I was unsure given that that hook would not return anything.
would you think it'd be fine to do just call the hook function without caring about the return value?

e.g.
```ts
const useGithubData = () => {
    // ...
}

const EditorComponentInner = React.memo() => {
  // ...
  useGithubData()
  // ...
  return <>...</>
}
```

Reply:
It's definitely worth making it honour the hooks rules, even if just to keep us safe from future refactoring introducing something that causes bugs.

Reply:
it's not about the return value, but about how hooks are implemented (there's a good description [here](https://stackoverflow.com/questions/54673188/how-does-react-implement-hooks-so-that-they-rely-on-call-order), it's up to you to decide how good an idea it was to implement hooks this way 😬).

Reply:
something like this? https://github.com/concrete-utopia/utopia/pull/2768/commits/0fdc67cce16a086d78d2e495125ca1f74961213f

Reply:
BTW hooks doesn't have to return anything, see e.g. `React.useEffect`